File: client/src/app/shared/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  type: string;
  brand: string;
  quantityInStock: number;
  status: 'Pending' | 'Approved' | 'Rejected' | 'Suspended';
  vendorId?: string;
  vendor?: {
    id: string;
    userName: string;
    firstName?: string;
    lastName?: string;
  };
  createdAt?: string;
  
  // Method to check if product is new (within 7 days)
  isNew?: () => boolean;
}

File: Core/Entities/Product.cs
using System;

namespace Core.Entities;

public class Product : BaseEntity
{
    public required string Name { get; set; }
    public required string Description { get; set; }
    public decimal Price { get; set; }
    public required string PictureUrl { get; set; }
    public required string Type { get; set; }
    public required string Brand { get; set; }
    public int QuantityInStock { get; set; }
    //vendor attributes
    public ProductStatus Status { get; set; } = ProductStatus.Approved; 
    public string? VendorId { get; set; } // optional
    public AppUser? Vendor { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Checks if the product is new (created within the last 7 days)
    /// </summary>
    /// <returns>True if the product is new, false otherwise</returns>
    public bool IsNew()
    {
        return CreatedAt >= DateTime.UtcNow.AddDays(-7);
    }
}
public enum ProductStatus
{
    Pending,    // Waiting for admin approval
    Approved,   // Approved by admin (visible to customers)
    Rejected,   // Rejected by admin
    Suspended   // Suspended by admin
}
